ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -427,28 +418,40 @@ - (BOOL)removeAllForUserId:(NSString * __nonnull)userId
                        (@{ @"operation" : @"removeAllForUserId:clientId:", @"clientId" : clientId, @"userId" : userId }),
                        @"userId: %@", userId);
     
-    BOOL deleteSuccessful = YES;
     NSArray* items = [self allItems:nil];
     
-    ADAuthenticationError* adError = nil;
-    
     for (ADTokenCacheItem * item in items)
     {
         if ([userId isEqualToString:[[item userInformation] userId]]
             && [clientId isEqualToString:[item clientId]])
         {
-            [self removeItem:item error:&adError];
-            if (adError)
+            if (![self removeItem:item error:error])
             {
-                deleteSuccessful = NO;
-                if (error)
-                {
-                    *error = adError;
-                }
+                return NO;
             }
         }
     }
-    return deleteSuccessful;
+    return YES;
+}
+
+- (BOOL)removeAllForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing  _Nullable *)error
+{
+    AD_LOG_WARN_DICT(([NSString stringWithFormat:@"Removing all items for user <%@>", userId]), nil,
+                     (@{ @"operation" : @"removeAllForClientId:", @"userId" : userId }), nil);
+
+    NSArray *items = [self allItems:nil];
+
+    for (ADTokenCacheItem *item in items)
+    {
+        if ([userId isEqualToString:[[item userInformation] userId]])
+        {
+            if (![self removeItem:item error:error])
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1066)